WordPress/gutenberg
# Enable the command palette feature everywhere in WordPress

## What?

Closed #58218. 

This PR introduce the ability to have the command palette everywhere in WordPress and not just inside of the new editor experience. 

## Why?
After talking to a few people who got excited about the command palette at WordCamp US we realized that in order to increase adoption of the command palette it needed to be available everywhere (so that it becomes an habit to reach out for it much like Spotlight Search on mac). 

## How?
This PR removes the need to add the component in both editor packages (edit-site and edit-post) and instead registers it at the global level. When the script is on the page, then the command palette is there as well. Yeah editor are then able to register custom commands as required.

Some global commands might need to be reworked to be available in a global context. Some of them at the moment do not redirect to a valid url. This PR is mainly to open a conversation. Feedback welcomed :)

## Testing Instructions
- Open the admin and press CMD+K (on mac) or CTRL+K (on windows) to validate that the command palette appear.

### Testing Instructions for Keyboard
Press the command palette keystroke combination to invoke the global component. 

## Screenshots or screencast 

<img width="1594" alt="CleanShot 2023-09-15 at 15 58 41@2x" src="https://github.com/WordPress/gutenberg/assets/1933681/0bc02bad-5e91-4a4e-9951-471abb85edc6">

<img width="1598" alt="CleanShot 2023-09-15 at 15 59 02@2x" src="https://github.com/WordPress/gutenberg/assets/1933681/e667dab2-55da-4c07-b82c-6c49dab94df2">
